src/Raven.Client/Documents/Session/Loaders/MultiLoaderWithInclude.cs

```diff
@@ -67,14 +68,59 @@ public ILoaderWithInclude<T> Include<TInclude>(Expression<Func<T, IEnumerable<st
             return Include(IncludesUtil.GetPrefixedIncludePath<TInclude>(path.ToPropertyPath(), _session.Conventions));
         }
 
+        /// <summary>
+        /// Includes a counter of the document to be loaded, by counter name
+        /// <param name="name">Name of the counter to include.</param>
+        /// </summary>
+        public ILoaderWithInclude<T> IncludeCounter(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+                throw new ArgumentNullException(nameof(name));
+
+            if (_counters == null)
+                _counters = new List<string>();
+
+            _counters.Add(name);
+            return this;
+        }
+
+        /// <summary>
+        /// Includes counters of the document to be loaded, by counters names
+        /// <param name="names">Names of the counters to include.</param>
+        /// </summary>
+        public ILoaderWithInclude<T> IncludeCounters(string[] names)
+        {
+            if (_counters == null)
+                _counters = new List<string>();
+
+            if (names == null)
+                throw new ArgumentNullException(nameof(names));
+
+            foreach (var name in names)
+            {
+                if (string.IsNullOrWhiteSpace(name))
+                    throw new InvalidOperationException("IncludeCounters(string[] names) : 'names' should not contain null or whitespace elements");
+                _counters.Add(name);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Includes all the counters of the document to be loaded
+        /// </summary>
+        public ILoaderWithInclude<T> IncludeCounters()
+        {
+            return IncludeCounters(new string[0]);
+        }
+
         /// <summary>
         /// Loads the specified ids.
         /// </summary>
         /// <param name="ids">The ids.</param>
         /// <returns></returns>
         public Dictionary<string, T> Load(params string[] ids)
         {
-            return _session.LoadInternal<T>(ids, _includes.ToArray());
+            return _session.LoadInternal<T>(ids, _includes.ToArray(), _counters?.ToArray());
```

Comment:
Why another allocation here? Why not pass the list?

Reply:
Probably rest of the API is done in the same way e.g. we are doing that with `_includes`

Reply:
leave it with ``string []`` for consistency, or change to List?

Reply:
keep it consistent